### PR TITLE
Support linear algebra library proposed by P1385R7 for inclusion to the C++ 23 Standard Library.

### PR DESCRIPTION
**Pull request by Twon**
Specify library name and version:  wg21-linear_algebra/0.7.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

I am submitting this library so that dependent libraries (such as https://github.com/mpusz/units) can use an official package. This library provides a reference implementation of the linear algebra library, which is proposed for inclusion into the C++ 23 Standard Library. I'm not the library author but have made numerous submissions to the library including the original custom Conan packaging and CI.

---

- [:white_check_mark:] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [:white_check_mark:] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [:white_check_mark:] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [:white_check_mark:] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wg21-linear_algebra/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LAConan(ConanFile):
+    name = "wg21-linear_algebra"
```
Comment (prince-chrismc) [inline recipes/wg21-linear_algebra/all/conanfile.py:9]: The name looks a little weird mixing `-` and `_` but it's valid

File: recipes/wg21-linear_algebra/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    homepage = "https://github.com/BobSteagall/wg21"
+    description = "Production-quality reference implementation of P1385: A proposal to add linear algebra support to the C++ standard library"
+    topics = ("linear-algebra", "multi-dimensional", "maths")
+    license = "BSD-3-Clause"
```
Comment (prince-chrismc) [inline recipes/wg21-linear_algebra/all/conanfile.py:13]: https://spdx.org/licenses/NCSA.html

```suggestion
    license = "NCSA"
```

File: recipes/wg21-linear_algebra/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        self.copy(pattern="*", dst="include",
+                  src=os.path.join(self._source_subfolder, "include"))
+        self.copy("*LICENSE*", dst="licenses", keep_path=False)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "wg21_linear_algebra"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "wg21_linear_algebra"
```
Comment (prince-chrismc) [inline recipes/wg21-linear_algebra/all/conanfile.py:70]: Default is to use names

```suggestion
```

**Review by prince-chrismc — COMMENTED** [review]
Looks very very good!

Please mark it ready for review

File: recipes/wg21-linear_algebra/all/conanfile.py (hunk 1 above)
Comment (Twon) [reply recipes/wg21-linear_algebra/all/conanfile.py:9]: Yes, agreed, but it matches the repo name.  Should this proposal be accepted in the C++ standard then there the wg21 prefix is there to distinguish between the proposal and a linear_algrebra accepted into the standard (which would presumably be std-linear_algebra)

**Comment by prince-chrismc** [comment]
> Please mark it ready for review

**Comment by SSE4** [comment]
@Twon any reason why it is still a draft? CI passes, so I wonder if you can make it ready

**Comment by Twon** [comment]
Apologies, I was waiting on changes in the main repo to migrate to Github actions.  I'll update the package index and submit once checks pass.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/wg21-linear_algebra/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "wg21_linear_algebra"
+        self.cpp_info.names["cmake_find_package_multi"] = "wg21_linear_algebra"
+        self.cpp_info.components["_wg21_linear_algebra"].names["cmake_find_package"] = "wg21_linear_algebra"
+        self.cpp_info.components["_wg21_linear_algebra"].names["cmake_find_package_multi"] = "wg21_linear_algebra"
+        self.cpp_info.components["_wg21_linear_algebra"].requires = ["mdspan::mdspan"]
```
Comment (SpaceIm) [inline recipes/wg21-linear_algebra/all/conanfile.py:73]: The 2 first lines are sufficient.

File: recipes/wg21-linear_algebra/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "8",
+            "clang": "8",
+            "apple-clang": "11"
+        }
+
+    def configure(self):
```
Comment (SpaceIm) [inline recipes/wg21-linear_algebra/all/conanfile.py:38]: it should be `validate()`

File: recipes/wg21-linear_algebra/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (SpaceIm) [inline recipes/wg21-linear_algebra/all/test_package/CMakeLists.txt:1]: `cxx_std_20` feature requires CMake >= 3.12

**Pull request merged**